xDrip/Scenes/Home/Home/HomeInteractor.swift

```diff
@@ -67,4 +68,9 @@ final class HomeInteractor: HomeBusinessLogic, HomeDataStore {
         let response = Home.GlucoseDataUpdate.Response(glucoseData: glucoseDataWorker.fetchGlucoseData())
         self.presenter?.presentGlucoseData(response: response)
     }
+    
+    private func updateGlucoseCurrentInfo() {
+        let response = Home.GlucoseCurrentInfo.Response(lastGlucoseReading: glucoseDataWorker.fetchGlucoseData().first)
```

Comment:
You're fetching all values and then take only the first one, which is a great performance issue. Create a separate method in `glucoseDataWorker` to fetch only the last entry.